frontend/src/utilities/__tests__/useWatchGroups.spec.tsx

```diff
@@ -1,41 +1,156 @@
+const useAppDispatchMock = jest.mocked(useAppDispatch);
+const dispatchMock = jest.fn();
+const updateGroupSettingsMock = jest.mocked(updateGroupsSettings);
+const mockEmptyGroupSettings = {
+  adminGroups: [],
+  allowedGroups: [],
+};
 
 describe('useWatchGroups', () => {
   it('should fetch groups successfully', async () => {
-    const mockEmptyGroupSettings = {
-      adminGroups: [],
+    const mockGroupSettings: GroupsConfig = {
+      adminGroups: [{ id: 1, name: 'odh-admins', enabled: true }],
       allowedGroups: [],
     };
+    updateGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    useAppDispatchMock.mockReturnValueOnce(dispatchMock);
+    fetchGroupSettingsMock.mockResolvedValue(mockGroupSettings);
+    const renderResult = testHook(useWatchGroups)();
+    expect(renderResult.result.current.updateGroups(mockGroupSettings)).toStrictEqual(undefined);
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockEmptyGroupSettings,
+      loaded: false,
+      isLoading: true,
+      isGroupSettingsChanged: false,
+      loadError: undefined,
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
+      setIsGroupSettingsChanged: expect.anything(),
+    });
+    await renderResult.waitForNextUpdate();
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(2);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockGroupSettings,
+      loaded: false,
+      isLoading: false,
+      isGroupSettingsChanged: false,
+      loadError: new TypeError('dispatch is not a function'),
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
+      setIsGroupSettingsChanged: expect.anything(),
+    });
+  });
+
+  it('should handle error', async () => {
+    useAppDispatchMock.mockReturnValue(dispatchMock);
+    fetchGroupSettingsMock.mockRejectedValue(new Error(`Error updating group settings`));
+    const renderResult = testHook(useWatchGroups)();
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockEmptyGroupSettings,
+      loaded: false,
+      isLoading: true,
+      isGroupSettingsChanged: false,
+      loadError: undefined,
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
+      setIsGroupSettingsChanged: expect.anything(),
+    });
+    await renderResult.waitForNextUpdate();
+    expect(fetchGroupSettingsMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).hookToStrictEqual({
+      groupSettings: mockEmptyGroupSettings,
+      loaded: false,
+      isLoading: false,
+      isGroupSettingsChanged: false,
+      loadError: Error('Error updating group settings'),
+      updateGroups: expect.anything(),
+      setGroupSettings: expect.anything(),
+      setIsGroupSettingsChanged: expect.anything(),
+    });
+  });
+
+  it('should test admin error', async () => {
```

Comment:
`errorAdmin` only has an effect on notifications but there are no notification assertions being made in this test.